### ISSUE TITLE
no-promise-as-boolean conflicts with no-return-await

### ISSUE THREAD

**Issue by buu700** [palantir/tslint#4851]
### Bug Report

- __TSLint version__: 5.19.0
- __TypeScript version__: 3.4.5
- __Running TSLint via__: (pick one) CLI

#### Reproduction using [TSLint Playground](https://palantir.github.io/tslint-playground)

The site is broken (not using my lint config as expected), but [here](https://palantir.github.io?saved=N4Igxg9gJgpiBc4IDsDOAXABAMwhTAvJgAoBOEAtgJaowB0pMqEANgG4wAU6pArjAEoA3AB1kkNFgBGAQ1KES5arQZNWHbn0GjkYgPR7MyCAFoADkpowTM1Cal4WMGckyQKZljKpoxEjJgYAJ5OAIwKuBB06AAWMMictkHimLIsLKiEAHypMumZAD4FuaTCYvqGxiaM6LykyDYA7t5Y7p7evuIoAcFOAEwReNFxCUkpaRnZufmYRZiJzVTScgJlyCAANEjI2FQA5gggBpgAmhC8bi6YFNBU2EGYsTBuKLt7dTLoVCiYcYyYjSWMUwQXO8ggjWQdAqmAAsjIANbPVB1Z6xT4gsGYABWzFcNEwbDyVCgG0wLigmDumIufHxyHQEBhECe8hoKKYZJ4DzAcTACJ8e0ecReaFYz1w8hgpHIpFQ0N0yGAYkwqswIhAfCcqA18EwytcaqNGqqFkoVhsdgc4pcuseWg2KqNapNphqdQaMkW6DtPH4TtVAF8xMH1oGgA).

#### TypeScript code being linted

This is the simplest case that illustrates the issue, but my actual code has other scenarios where `foo` might be something like `SafeUrl|string|undefined` while `bar` is `SafeUrl` or the expression might use `&&` instead of `||`.

```ts
const foo = Promise.resolve(true);
const bar = Promise.resolve(true);

// no-promise-as-boolean complains
const style1 = foo.then(async balls => balls || bar);

// no-return-await complains
const style2 = foo.then(async balls => balls || (await bar));
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-promise-as-boolean": true,
        "no-return-await": true
    }
}
```

#### Actual behavior

I have to add a lint exception comment.

#### Expected behavior

At least one of these two styles should be accepted by both rules.

**Comment by JoshuaKGoldberg** [comment]
Hmm, I'm not sure this issue is very actionable. Neither rule is enabled in the recommended ruleset, and other rules end up conflicting with each other by nature of some being more opinionated than others. What would you like to see here @buu700?

**Comment by buu700** [comment]
How about an option to `no-promise-as-boolean` called something like `"allow-when-it's-the-last-part-of-a-return-statement"`? 99% of the time this rule seems really helpful and not at all contradictory with `no-return-await`.

**Comment by JoshuaKGoldberg** [comment]
Yeah, that seems reasonable. Though a slightly shorter name would be swell... 😜How about `allow-return-ending`?

Obligatory note for #4534 - soon, this will all be gone anyway.

**Comment by buu700** [comment]
lol, that sounds good to me.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
